src/autorenew-invite/autorenew-invite.controller.js

```diff
@@ -0,0 +1,83 @@
+import _ from 'lodash';
+import moment from 'moment';
+
+export default class {
+  /* ngInject */
+  constructor(
+    $q,
+    OvhApiMe,
+    ovhPaymentMethod,
+    ovhUserPref,
+    WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW,
+  ) {
+    this.$q = $q;
+    this.OvhApiMe = OvhApiMe;
+    this.ovhPaymentMethod = ovhPaymentMethod;
+    this.ovhUserPref = ovhUserPref;
+
+    this.WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW = WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW;
+  }
+
+  $onInit() {
+    this.preferenceKey = `${this.productType.toUpperCase()}_AUTORENEW_STOP_BOTHER`;
+
+    return this.isAutorenewAllowed()
+      .then(() => (this.isAutorenewAllowed ? this.getDisplayConditions() : this.$q.when()));
+  }
+
+  getDisplayConditions() {
+    return this.getProductAutorenewBotherPreferences()
+      .then(() => this.hasPaymentMean());
+  }
+
+  hasPaymentMean() {
+    return this.ovhPaymentMethod.getAllPaymentMethods()
+      .then((paymentMethods) => {
+        this.hasPaymentMean = !_.isEmpty(paymentMethods);
```

Comment:
```suggestion
         this.hasPaymentMean = _.isArray(paymentMethods) && !_.isEmpty(paymentMethods);
```

Lodash doesn't like defensive pattern (have you seen any `ul` in Lodash ??) so if `paymentMethods` is a `string`, you would have an unexpected result